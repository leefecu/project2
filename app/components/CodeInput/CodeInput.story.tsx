import React from 'react';
import { Text, View } from 'react-native';

import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

import CodeInput from './CodeInput';
import Styles from '../../Stories.styles';

storiesOf('Component | CodeInput', module).add('Primary', () => {
  return (
    <View style={Styles.containerWithBG}>
      <View style={Styles.row}>
        <Text style={Styles.title}>Default</Text>
        <CodeInput code="" onUpdate={action('update number')} />
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>with Code</Text>
        <CodeInput code="12345" onUpdate={action('update number')} />
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>successful</Text>
        <CodeInput code="12345" onUpdate={action('update number')} success={true} />
      </View>
    </View>
  );
});
