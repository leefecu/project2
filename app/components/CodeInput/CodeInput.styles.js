import { StyleSheet } from 'react-native';

import Colors from '../../themes/Colors';
import Metrics from '../../themes/Metrics';

export default StyleSheet.create({
  container: {
    backgroundColor: Colors.transparent,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-evenly',
  },
  codeInput: {
    backgroundColor: Colors.white,
    borderColor: Colors.white,
    borderRadius: Metrics.inputBorderRadius,
    borderWidth: 1,
    color: Colors.primaryDark,
    fontSize: Metrics.Fonts.h4,
    fontWeight: 'bold',
    fontFamily: 'CircularPro-Bold',
    textAlign: 'center',
    marginHorizontal: Metrics.smallerMargin,
    height: 68,
    width: 48,

    shadowColor: Colors.black,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  hiddenInput: {
    height: 0,
  },
  codeWrapper: {
    position: 'relative',
  },
  tick: {
    position: 'absolute',
    top: 26,
    left: 20,
  },
});
