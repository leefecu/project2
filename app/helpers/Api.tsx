import get from 'lodash/get';
import reject from 'lodash/reject';
import Config from 'react-native-config';
import compareVersions from 'compare-versions';
import { Platform } from 'react-native';

import * as Storage from './Storage';
import { createExpiryNotification } from './Notification';
import { convertStashToActivePAck } from './Transformation';
import API from '../services/Api';

import { SYSTEM_RESPONSE } from '../constants/Shared';

const api = API.create();

export const purchasePack = async (packetId: string, promoId?: string) => {
  const response = await api.purchasePack(packetId, promoId);
  const currentActivePack = await Storage.getActivePack();
  if (response.ok) {
    const balance = get(response, 'data.balance');
    const purchasedPack = get(response, 'data.purchasedPack');
    const activePack = currentActivePack
      ? {
          ...currentActivePack,
          endTime: purchasedPack.endTime,
        }
      : purchasedPack;

    createExpiryNotification(purchasedPack.endTime);
    await Storage.setActivePackAndBalance(activePack, balance);
    return response;
  }
  return response;
};

export const activateStash = async (stash: TStash) => {
  const response = await api.activateStash(stash.packetInstanceId);
  if (response.ok) {
    const currentActivePack = await Storage.getActivePack();
    const currentStash = await Storage.getStash();
    const newActivePack = convertStashToActivePAck(stash, currentActivePack);
    const newStash = reject(
      currentStash,
      (target: TStash) => target.packetInstanceId === stash.packetInstanceId,
    );

    createExpiryNotification(newActivePack.endTime);
    await Storage.saveActivePackAndStash(newActivePack, newStash);
    return response;
  }
  return response;
};

export const checkSystemInfo = async (): Promise<SYSTEM_RESPONSE> => {
  const response = await api.getSystemInfo();
  if (response.ok) {
    const maintenance: boolean = get(response, 'data.maintenance', false);
    if (maintenance) {
      return SYSTEM_RESPONSE.MAINTENANCE;
    }
    const minVersion = get(response, `data.${Platform.OS}`);
    const appVersion = Config.APP_VERSION;

    return compareVersions.compare(appVersion, minVersion, '>=')
      ? SYSTEM_RESPONSE.CONTINUE
      : SYSTEM_RESPONSE.UPGRADE;
  }
  return SYSTEM_RESPONSE.CONTINUE;
};
