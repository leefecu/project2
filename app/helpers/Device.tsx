import { NativeModules, Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import CarrierInfo from 'react-native-carrier-info';
import RNFS from 'react-native-fs';

const NZ_MCC = '530';
const NZ_COUNTRY_CODE = 'nz';
const IS_IOS = Platform.OS === 'ios';

const NetworkInfo = NativeModules.RNNetworkInfo;

export const isSimulator = async (): Promise<boolean> => {
  const isEmulator = await DeviceInfo.isEmulator();
  return isEmulator;
};

export const isSimIn = async () => {
  // react-native-carrier-info does not work in simulator
  if (await isSimulator()) {
    return true;
  }

  try {
    // mobileCountryCode method fails when no SIM card in the device
    await CarrierInfo.mobileCountryCode();
    return true;
  } catch (e) {
    return false;
  }
};

export const checkForRoaming = async (debug: boolean = false): Promise<boolean> => {
  // react-native-carrier-info does not work in simulator
  if (await isSimulator()) {
    return debug;
  }

  try {
    if (Platform.OS === 'ios') {
      const operatorPlist = await NativeModules.NetworkInfo.getOperatorPlist();
      const mcc = operatorPlist.match(/\d+\/carrier.plist/g);
      if (mcc && mcc.length > 0) {
        return mcc[0].substr(0, 3) !== NZ_MCC;
      }
      return false;
    }
    const countryCode = await NetworkInfo.isoCountryCode();
    return countryCode !== NZ_COUNTRY_CODE;
  } catch (e) {
    return false;
  }
};

const majorVersionIOS: number = parseInt(`${Platform.Version}`, 10);

export const isAutoFillSupported: boolean = IS_IOS && majorVersionIOS >= 12;

// Figured out how current(old) DataClock app stores msisdn
// it's stored in xml file
// shared_prefs/nz.co.twodegreesmobile.comptelfwd.smpl.Account.xml
const msisdnFileName = 'nz.co.twodegreesmobile.comptelfwd.smpl.Account.xml';
const msisdnFolderName = 'shared_prefs';
const msisdnRegex = /com.comptel.smpl.interaction.Account.PHONE_NUMBER_KEY">\d+<\/string/g;

export const getExistingMsisdn = async (): Promise<TMigrationData> => {
  if (IS_IOS) {
    try {
      const msisdn = await NativeModules.GetNumber.getMsisdn();
      return { number: msisdn, upgrade: !!msisdn };
    } catch (e) {
      return { number: '', upgrade: false };
    }
  }
  // This method only works with Android
  try {
    // e.g. RNFS.DocumentDirectoryPath = '/data/user/0/nz.co.twodegreesmobile.comptelfwd.smpl/files',
    const paths = RNFS.DocumentDirectoryPath.split('/');
    // need to remove "/files" and replace with "shared_prefs"
    paths[paths.length - 1] = msisdnFolderName;
    const path = paths.join('/');
    const xmlPath = `${path}/${msisdnFileName}`;
    const existingAccount = await RNFS.readFile(xmlPath);
    const match = msisdnRegex.exec(existingAccount);
    if (match && match.length > 0) {
      const msisdn = match[0]
        .replace('com.comptel.smpl.interaction.Account.PHONE_NUMBER_KEY">', '')
        .replace('</string', '');

      return { number: msisdn, upgrade: true };
    }
    return { number: '', upgrade: false };
  } catch (e) {
    return { number: '', upgrade: false };
  }
};
