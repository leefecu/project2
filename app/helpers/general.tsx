import { NavigationParams, NavigationScreenProp, NavigationState } from 'react-navigation';

import { ERROR_CODE } from '../constants/Messages';
import { SYSTEM_RESPONSE } from '../constants/Shared';

import * as API from '../helpers/Api';

export const noop = () => {};

interface TTestProp {
  testID: string;
  accessibilityLabel: string;
}
export const testProps = (id: string): TTestProp => ({
  testID: id,
  accessibilityLabel: id,
});

export const shouldLogout = (status: number | undefined, errorCode: ERROR_CODE): boolean =>
  status === 401 ||
  status === 403 ||
  errorCode === ERROR_CODE.NOT_REGISTERED ||
  errorCode === ERROR_CODE.INVALID_PLAN ||
  errorCode === ERROR_CODE.QD_ISSUE;

export const systemCheck = async (
  navigation: NavigationScreenProp<NavigationState, NavigationParams>,
) => {
  const systemInfo = await API.checkSystemInfo();
  if (systemInfo === SYSTEM_RESPONSE.CONTINUE) {
    return;
  }
  navigation.navigate(systemInfo === SYSTEM_RESPONSE.MAINTENANCE ? 'Maintenance' : 'Upgrade');
};
