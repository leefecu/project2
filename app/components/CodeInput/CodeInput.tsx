import React, { createRef, PureComponent, RefObject } from 'react';
import {
  NativeSyntheticEvent,
  Platform,
  TextInput,
  View,
  TextInputKeyPressEventData,
} from 'react-native';
import get from 'lodash/get';
import times from 'lodash/times';

import { isAutoFillSupported } from '../../helpers/Device';
import { noop, testProps } from '../../helpers/general';

import Styles from './CodeInput.styles';

import TickSVG from '../../images/tick.svg';

interface DefaultProps {
  invalid?: boolean;
  code: string;
  success?: boolean;
  onBlur?: () => void;
  onUpdate: (number: string) => void;
}

interface Props extends DefaultProps {}

export const OPT_DIGIT_COUNT = 5;
export const EMPTY_PLACEHOLDER = 'X';

class Input extends PureComponent<Props> {
  autoFillInput: RefObject<TextInput>;

  codes: Array<RefObject<TextInput>>;

  static defaultProps: DefaultProps = {
    invalid: false,
    code: '',
    success: false,
    onBlur: noop,
    onUpdate: noop,
  };

  constructor(props: Props) {
    super(props);
    this.codes = [];

    this.autoFillInput = createRef<TextInput>();

    times(OPT_DIGIT_COUNT, index => {
      this.codes[index] = createRef<TextInput>();
    });
  }

  getCode(position: number) {
    const { code, success } = this.props;
    if (success) {
      return '';
    }
    return code.length > position
      ? code[position] === EMPTY_PLACEHOLDER
        ? ''
        : code[position]
      : '';
  }

  handleCodeChange = (position: number, value: string) => {
    const { code, onUpdate } = this.props;
    if (value) {
      const newCode: string =
        code.substring(0, position) + value + code.substring(position + 1, OPT_DIGIT_COUNT);
      onUpdate(newCode);
      if (position < OPT_DIGIT_COUNT - 1) {
        const nextCode = get(this.codes, `${[position + 1]}.current`, null);
        if (nextCode) {
          nextCode.focus();
        }
      }
    }
  };

  handleDeleteCode = (position: number, key: string) => {
    const { code, onUpdate } = this.props;
    if (key === 'Backspace' && position >= 0) {
      const newCode: string =
        code.substring(0, position) +
        EMPTY_PLACEHOLDER +
        code.substring(position + 1, OPT_DIGIT_COUNT);
      onUpdate(newCode);
      if (this.getCode(position) === '' && position > 0) {
        const prevCode = get(this.codes, `${[position - 1]}.current`, null);
        if (prevCode) {
          prevCode.focus();
        }
      }
    }
  };

  handleAutoFill = (code: string) => {
    const { onUpdate } = this.props;
    if (code.length === OPT_DIGIT_COUNT) {
      onUpdate(code);
    }
  };

  handleAutoFillInputKeyPress = ({
    nativeEvent: { key },
  }: NativeSyntheticEvent<TextInputKeyPressEventData>) => {
    const { code, onUpdate } = this.props;
    const secondDigit = get(this.codes, `${[1]}.current`, null);

    if (secondDigit && Number(key)) {
      const newCode = key + code.substring(1, OPT_DIGIT_COUNT);
      onUpdate(newCode);
      secondDigit.focus();
    }
  };

  render() {
    const { code, invalid, onBlur, onUpdate, success, ...props } = this.props;
    return (
      <View style={Styles.container} {...testProps('One Time Passcode Input Field')}>
        {times(OPT_DIGIT_COUNT, index => (
          <View key={index} style={Styles.codeWrapper}>
            <TextInput
              ref={this.codes[index]}
              autoFocus={Platform.OS !== 'android' && !isAutoFillSupported && index === 0}
              keyboardType="number-pad"
              maxLength={1}
              onBlur={onBlur}
              onChangeText={value => this.handleCodeChange(index, value)}
              onKeyPress={e => this.handleDeleteCode(index, e.nativeEvent.key)}
              returnKeyType="next"
              style={Styles.codeInput}
              value={this.getCode(index)}
              {...props}
            />
            {success && (
              <View style={Styles.tick}>
                <TickSVG />
              </View>
            )}
          </View>
        ))}
        {/* 
        Trick here is to display hidden TextInput to trigger keyboard up
        and display OTC once SMS is received
        This only works for ios12 or above
        Applying textContentType="oneTimeCode" doesn't really work  with above TextInput
        as it's composed of 5 individual TextInput
        */}
        {isAutoFillSupported && (
          <TextInput
            ref={this.autoFillInput}
            keyboardType="number-pad"
            autoFocus={true}
            style={Styles.hiddenInput}
            textContentType="oneTimeCode"
            onChangeText={this.handleAutoFill}
            onKeyPress={this.handleAutoFillInputKeyPress}
            value=""
          />
        )}
      </View>
    );
  }
}

export default Input;
