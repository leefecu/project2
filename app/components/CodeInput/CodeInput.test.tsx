/**
 * @jest-environment jsdom
 */
import React from 'react';
import { create } from 'react-test-renderer';

import CodeInput from './CodeInput';

import { noop } from '../../helpers/general';

jest.mock('react-native-device-info', () => {
  return {
    isEmulator: jest.fn(),
  };
});

describe('CodeInput component', () => {
  test('render default codeInput', () => {
    const button = create(<CodeInput code="" onUpdate={noop} />);
    expect(button.toJSON()).toMatchSnapshot();
  });
  test('render codeInput with number', () => {
    const button = create(<CodeInput code="12345" onUpdate={noop} />);
    expect(button.toJSON()).toMatchSnapshot();
  });
  test('render codeInput with tick', () => {
    const button = create(<CodeInput code="12345" onUpdate={noop} success={true} />);
    expect(button.toJSON()).toMatchSnapshot();
  });
});
