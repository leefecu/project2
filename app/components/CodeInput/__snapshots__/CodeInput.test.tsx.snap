// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeInput component render codeInput with number 1`] = `
<View
  accessibilityLabel="One Time Passcode Input Field"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flexDirection": "row",
      "justifyContent": "space-evenly",
    }
  }
  testID="One Time Passcode Input Field"
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={true}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value="1"
    />
  </View>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value="2"
    />
  </View>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value="3"
    />
  </View>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value="4"
    />
  </View>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value="5"
    />
  </View>
</View>
`;

exports[`CodeInput component render codeInput with tick 1`] = `
<View
  accessibilityLabel="One Time Passcode Input Field"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flexDirection": "row",
      "justifyContent": "space-evenly",
    }
  }
  testID="One Time Passcode Input Field"
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={true}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "left": 20,
          "position": "absolute",
          "top": 26,
        }
      }
    >
      <SVGMock />
    </View>
  </View>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "left": 20,
          "position": "absolute",
          "top": 26,
        }
      }
    >
      <SVGMock />
    </View>
  </View>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "left": 20,
          "position": "absolute",
          "top": 26,
        }
      }
    >
      <SVGMock />
    </View>
  </View>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "left": 20,
          "position": "absolute",
          "top": 26,
        }
      }
    >
      <SVGMock />
    </View>
  </View>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "left": 20,
          "position": "absolute",
          "top": 26,
        }
      }
    >
      <SVGMock />
    </View>
  </View>
</View>
`;

exports[`CodeInput component render default codeInput 1`] = `
<View
  accessibilityLabel="One Time Passcode Input Field"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flexDirection": "row",
      "justifyContent": "space-evenly",
    }
  }
  testID="One Time Passcode Input Field"
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={true}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={false}
      keyboardType="number-pad"
      maxLength={1}
      onBlur={[Function]}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      rejectResponderTermination={true}
      returnKeyType="next"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#004071",
          "elevation": 5,
          "fontFamily": "CircularPro-Bold",
          "fontSize": 24,
          "fontWeight": "bold",
          "height": 68,
          "marginHorizontal": 4,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "textAlign": "center",
          "width": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
